Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/CudaUtils.cuh

```diff
@@ -0,0 +1,32 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <sstream>
+
+#include <cuda_runtime_api.h>
+
+namespace Acts::detail {
+
+inline void cudaAssert(cudaError_t code, const char *file, int line) {
+  if (code != cudaSuccess) {
+    std::stringstream ss;
+    ss << "CUDA error: " << cudaGetErrorString(code) << ", " << file << ":"
+       << line;
+    throw std::runtime_error(ss.str());
+  }
+  cudaDeviceSynchronize();
```

Comment:
Note that while this is probably okay for now, this will kill any asynchronous operations, as the result checking of those operations will implicitly add a synchronization point.

Reply:
Good catch, actually, I guess there is not a lot benefit from this synchronization point here, right? I think its a leftover from a state where the chode was not async at all

Reply:
Yeah, I'd remove it.

Reply:
Although it _might_ be useful to synchronize before throwing the exception... I'm not sure. :thinking: